web-api/src/personalization/routes.rs

```diff
@@ -138,61 +139,96 @@ const fn default_include_properties() -> bool {
 
 /// Represents personalized documents query params.
 #[derive(Debug, Deserialize)]
-struct PersonalizedDocumentsQuery {
+struct UnvalidatedPersonalizedDocumentsQuery {
     count: Option<usize>,
     published_after: Option<DateTime<Utc>>,
+    filter: Option<String>,
```

Comment:
we need to deserialize the filter as an `Option<String>` first, because the query params can be url encoded, which would be break deserializing directly as a `Option<Filter>`. hence i adopted the validation approach from the other endpoints here.

maybe in a follow up, we should also deprecate the query params and move them to a request body.